docs/pages/principles/testing.md

```diff
@@ -0,0 +1,94 @@
+## Unit Tests
+
+### Advantages of unit testing:
+
+- Unit tests ensure that the code, as written, is correct, and executes
+  properly.
+- They communicate the intention of the creator of the code, how the code is
+  expected to behave, in its expected use-case.
+
+- Writing unit tests can reveal weakensses in our implementations, and lead us
+  to better design decisions:
+  - If the test requires excessive setup, the unit may be dependent on too many
+    external variables.
```

Comment:
```suggestion
  - If the test requires excessive setup, the unit may be dependent on too many
    external variables. You might benefit from refactoring the code.
```

I think it would help here to make sure the reader knows the problme is in the code ("unit" here), not the test. That's clear in the last point, but I'd make sure it's obvious here too. Maybe this could be repeated every bullet point to drive it in.